apps/heft/src/plugins/JestPlugin/JestTypeScriptDataFile.ts

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+import { JsonFile } from '@rushstack/node-core-library';
+import { ITypeScriptConfiguration } from '../../stages/BuildStage';
+
+/**
+ * Schema for jest-typescript-data.json
+ */
+export interface IJestTypeScriptDataFileJson {
+  /**
+   * The "emitFolderPathForJest" from .heft/typescript.json
+   */
+  emitFolderPathForJest: string;
+}
+
+/**
+ * Manages loading/saving the "jest-typescript-data.json" data file.  This file communicates
+ * configuration information from Heft to jest-build-transform.js.  The jest-build-transform.js script gets
+ * loaded dynamically by the Jest engine, so it does not have access to the normal HeftConfiguration objects.
```

Comment:
We actually pass the `HeftConfiguration` object to the Jest reporter (see [here](https://github.com/microsoft/rushstack/blob/master/apps/heft/src/plugins/JestPlugin/JestPlugin.ts#L36)). Maybe we can do something similar here?

Reply:
@iclanton  The difference is that reporters are runtime objects -- they wire up Jest's output to Heft's terminal.  It would not be possible for `jest-cli` to use that reporter (as currently written).

Whereas transforms are config objects -- `jest-cli` could invoke these transforms without any trouble.  Today we do not actually support `jest-cli` (but it is technically possible).  

And in general, as much as possible, we should try to make the Jest configuration explicit via config files rather than constructing it invisibly at runtime.